packages/api/src/dpi.ts

```diff
@@ -33,6 +55,48 @@ class LogicalSize {
   toPhysical(scaleFactor: number): PhysicalSize {
     return new PhysicalSize(this.width * scaleFactor, this.height * scaleFactor)
   }
+
+  /**
+   * Converts this size into IPC-compatible value, so it can be
+   * deserialized correctly on the Rust side using `tauri::LogicalSize` struct.
+   * @example
+   * ```typescript
+   * import { LogicalSize } from '@tauri-apps/api/dpi';
+   * import { invoke } from '@tauri-apps/api/core';
+   *
+   * const size = new LogicalSize(400, 500);
+   * await invoke("do_something_with_size", { size: size.toIPC() })
+   * ```
+   *
+   * @since 2.0.0
+   */
+  [SERIALIZE_TO_IPC_FN]() {
```

Comment:
I like the whole toIPC approach (makes the `__CHANNEL__` thingy available to anyone) but this is actually a breaking change - existing commands relying on PhysicalPosition / PhysicalSize / LogicalPosition / LogicalSize cannot deserialize this new format, so we should revert these - if you explictly want a generic Position or Size type we need to create a wrapper type just like Rust has one

Reply:
that makes sense, I have added `Position` and `Size` classes and reverted breaking changes.